src/PropertySuggester/Suggesters/SimpleSuggester.php

```diff
@@ -73,38 +78,58 @@ public function setInitialSuggestions( array $initialSuggestionIds ) {
 	}
 
 	/**
-	 * @param int[] $propertyIds
-	 * @param string[] $idTuples
+	 * @param float $classifyingConditionWeight
+	 */
+	public function setClassifyingConditionWeight( $classifyingConditionWeight ) {
+		$this->classifyingConditionWeight = $classifyingConditionWeight;
+	}
+
+	/**
+	 * @param array[] $idTuples array of ( int Property id, int Item id) tuples
 	 * @param int $limit
 	 * @param float $minProbability
 	 * @param string $context
+	 *
 	 * @throws InvalidArgumentException
+	 *
 	 * @return Suggestion[]
 	 */
-	protected function getSuggestions( array $propertyIds, array $idTuples, $limit, $minProbability, $context ) {
+	private function getSuggestions( array $idTuples, $limit, $minProbability, $context ) {
 		if ( !is_int( $limit ) ) {
 			throw new InvalidArgumentException( '$limit must be int!' );
 		}
 		if ( !is_float( $minProbability ) ) {
 			throw new InvalidArgumentException( '$minProbability must be float!' );
 		}
-		if ( !$propertyIds ) {
+		if ( !$idTuples ) {
 			return $this->initialSuggestions;
 		}
 
+		$propertyIds = array_map( function( array $tuple ) {
+			return $tuple[0];
+		}, $idTuples );
 		$excludedIds = array_merge( $propertyIds, $this->deprecatedPropertyIds );
 		$count = count( $propertyIds );
 
-		$dbr = $this->lb->getConnection( DB_SLAVE );
-		if ( empty( $idTuples ) ){
-			$condition = 'pid1 IN (' . $dbr->makeList( $propertyIds ) . ')';
+		$tupleConditions = [];
+		$hasClassifyingPropertyIds = false;
+		foreach ( $idTuples as $tuple ) {
+			$tupleConditions[] = $this->buildTupleCondition( $tuple[0], $tuple[1] );
+			$hasClassifyingPropertyIds = $hasClassifyingPropertyIds || $tuple[1] > 0;
 		}
-		else{
-			$condition = $dbr->makeList( $idTuples, LIST_OR );
+
+		$dbr = $this->lb->getConnection( DB_REPLICA );
+		if ( $hasClassifyingPropertyIds && $this->classifyingConditionWeight != 0.5 ) {
```

Comment:
I forgot about this… I actually wanted to replace this with a safer float comparison.